apps/api/src/app/integrations/usecases/get-webhook-support-status/get-webhook-support-status.usecase.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { IEmailProvider, ISmsProvider } from '@novu/stateless';
+import { MailFactory, SmsFactory, ISmsHandler, IMailHandler } from '@novu/application-generic';
+import { GetWebhookSupportStatusCommand } from './get-webhook-support-status.command';
+
+@Injectable()
+export class GetWebhookSupportStatus {
+  public readonly mailFactory = new MailFactory();
+  public readonly smsFactory = new SmsFactory();
+  private provider: IEmailProvider | ISmsProvider;
+
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: GetWebhookSupportStatusCommand): Promise<boolean> {
+    const { providerId, channel } = command;
+
+    const integration: IntegrationEntity = await this.getIntegration(command);
+
+    if (!integration) {
+      throw new NotFoundException(`Integration for ${providerId} was not found`);
+    }
+
+    this.createProvider(integration, providerId, channel);
+
+    if (!this.provider.getMessageId || !this.provider.parseEventBody) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private async getIntegration(command: GetWebhookSupportStatusCommand): Promise<IntegrationEntity> {
+    return await this.integrationRepository.findOne({
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+      providerId: command.providerId,
+      channel: command.channel,
+    });
+  }
+
+  private getHandler(integration: IntegrationEntity, channel: 'email' | 'sms'): ISmsHandler | IMailHandler {
+    switch (channel) {
+      case 'sms':
+        return this.smsFactory.getHandler(integration);
+      default:
+        return this.mailFactory.getHandler(integration);
+    }
+  }
+
+  private createProvider(integration: IntegrationEntity, providerId: string, type: 'sms' | 'email') {
+    const handler = this.getHandler(integration, type);
```

Comment:
The integration can retrieve the information about the property channel so I keep supporting my suggestion of getting rid of it from the URL path.
https://github.com/novuhq/novu/blob/2d3050a13b5dfc77b0d66e34ed2eac8a25299631/libs/dal/src/repositories/integration/integration.entity.ts#L33